app/Http/Controllers/Auth/LoginController.php

```diff
@@ -36,161 +28,78 @@ class LoginController extends BaseController
      */
     public function __construct()
     {
-        $this->middleware('guest')->except('showLoginForm', 'loginSecondary', 'logout');
+        $this->provider = new VatsimOAuthController();
```

Comment:
Because this is a controller, you should be able to inject this class as constructor param and have the service container resolve it for you. Having said that, you might need to bind a singleton. Give it a go!

Reply:
Done!